homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -985,12 +989,24 @@ async def async_unjoin_player(self) -> None:
         await self.async_beolink_leave()
 
     # Custom actions:
-    async def async_beolink_join(self, beolink_jid: str | None = None) -> None:
+    async def async_beolink_join(
+        self, beolink_jid: str | None = None, source_id: str | None = None
+    ) -> None:
         """Join a Beolink multi-room experience."""
+        # Touch to join
         if beolink_jid is None:
             await self._client.join_latest_beolink_experience()
-        else:
+        # Join a peer
+        elif beolink_jid and source_id is None:
             await self._client.join_beolink_peer(jid=beolink_jid)
+        # Join a peer and select specific source
+        elif beolink_jid and source_id:
+            # Home Assistant does not support the "raw" Beolink Converter NL/ML source ids in strings.json
+            # So they have the be transformed to be compatible and now have to be capatilized in order to work
+            if BL_CONVERTER_PREFIX in source_id:
```

Comment:
Two things:

1. Explicitly check for the prefix (vs. a substring):
```python
if source_id.startswith(BL_CONVERTER_PREFIX):
```

2. Please add a test to cover this unique functionality.

Reply:
See comment below first. Looking to better understand why this is necessary. :)

Reply:
Yes, I'll add a test for that.